### PR TITLE
Handle Interrupted events in writeDataFromProducer()

### PR DESCRIPTION
**Pull request by ikesyo**
Same as ReactiveCocoa/ReactiveCocoa#2010.

### CODE REVIEW

File: ReactiveTask/Task.swift [hunk 1]
```diff
@@ -175,6 +175,8 @@ private final class Pipe {
 					}
 				}, completed: {
 					dispatch_io_close(channel, 0)
+				}, interrupted: {
```
Comment (robrix) [inline ReactiveTask/Task.swift]: This should close the channel, I think?

Comment (ikesyo) [inline ReactiveTask/Task.swift]: Sending Interrupted event disposes the disposable I think, so `dispatch_io_close(channel, DISPATCH_IO_STOP)` would be called finally?

Comment (robrix) [inline ReactiveTask/Task.swift]: Okay.

**Comment by robrix** [comment]
I’d like to get @jspahrsummers or @keithduncan to take a look at this if they could :pray:

**Comment by jspahrsummers** [comment]
Seems reasonable, but AFAICT this won't affect any current behaviors in practice. Thanks for the PR! :sparkles:

**Pull request merged**